frontend/src/metabase/actions/components/ActionViz/ActionDashcardSettings.unit.spec.tsx

```diff
@@ -155,6 +194,23 @@ describe("ActionViz > ActionDashcardSettings", () => {
     expect(screen.getByText("Action Parameter 2")).toBeInTheDocument();
   });
 
+  it("shows hidden badge for hidden and required field", () => {
+    setup({
+      dashcard: actionDashcardWithActionWithHiddenFields,
+    });
+
+    const formSection = screen.getByTestId(
+      `parameter-form-section-${actionParameter3.id}`,
+    );
+
+    expect(formSection).toBeInTheDocument();
+    expect(within(formSection).getByText(/is required/i)).toBeInTheDocument();
+    expect(within(formSection).getByText(/hidden/i)).toBeInTheDocument();
+    expect(
+      within(formSection).getByRole("button", { name: /select a value/i }),
+    ).toBeInTheDocument();
+  });
```

Comment:
I don't see a test that makes sure you cannot submit the mapping form when one of the fields is required

Reply:
added a test